scalyr_agent/builtin_monitors/kubernetes_events_monitor.py

```diff
@@ -213,6 +214,18 @@ class KubernetesEventsMonitor( ScalyrMonitor ):
        ```
     """
 
+    ENVIRONMENT_AWARE_CONFIG_PARAMS = {
```

Comment:
Why not modify the `define_config_option` method to accept an optional parameter specifying what environment variable it defaults to?  That way all information about the configuration parameter is located together.

Reply:
My initial design was actually an optional param in `define_config_option` and there are certainly benefits of colocating it with the rest of the param definition.  However, I later realized 2 things:

1) we should adopt a uniform style for both global and monitor-level (ie both centralized map or else both during `define_config_option` or `__verify_or_set_optional_xxx`.

2) As the number of total config options balloons, there's benefit in being able to quickly identify the complete list of environment aware params.  I felt it was hard to track them when spread out.  

2) For testing.  Without this, I think the individual calls to `__verify_or_set_optional_xxx` would have to build this map at runtime anyway.

WDYT? If you still feel strongly about this,  I can look into changing it.

Reply:
I do agree with your point number `1`... we should move to a `define_config_option` for the global parameters.  However, I'm not sure if you want to do all of that work in this PR.  Adding the environment aware arguments to `__verify_or_set..` is at least a step towards that.

As for point `2`, I'm not sure if I agree.  I can't really see me saying "what are the environment aware params?".  It's much more likely that I will ask myself "for this particular param, what is it's default and can I set it via an environment variable?".   If you are worried that we will need to know the complete list to avoid collision.. I would suggest we should solve that through code.  If we attempt to define two options with the same environment variable, throw an assertion.

I do agree that the `__verify_or_set_optional_xxx` will have to build the map at runtime, but that's fine.

I do think we should make this change now.  I think we want to be going in the right eventual direction and this is a step away from that.  WDYT?

Reply:
All fair points.  I will change.